collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -161,26 +180,133 @@ func (r *telemetryAPIReceiver) httpHandler(w http.ResponseWriter, req *http.Requ
+		} else {
+			r.logger.Error("error parsing time", zap.Error(err))
+			return plog.Logs{}, err
+		}
+		if el.Type == string(telemetryapi.Function) || el.Type == string(telemetryapi.Extension) {
+			if record, ok := el.Record.(map[string]interface{}); ok {
+				// in JSON format https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#telemetry-api-function
+				if timestamp, ok := record["timestamp"].(string); ok {
+					if t, err := time.Parse(timeFormatLayout, timestamp); err == nil {
+						logRecord.SetTimestamp(pcommon.NewTimestampFromTime(t))
+					} else {
+						r.logger.Error("error parsing time", zap.Error(err))
+						return plog.Logs{}, err
+					}
+				}
+				if level, ok := record["level"].(string); ok {
+					level = strings.ToUpper(level)
+					switch level {
+					case "TRACE":
+						logRecord.SetSeverityNumber(1)
+					case "TRACE2":
+						logRecord.SetSeverityNumber(2)
+					case "TRACE3":
+						logRecord.SetSeverityNumber(3)
+					case "TRACE4":
+						logRecord.SetSeverityNumber(4)
+					case "DEBUG":
+						logRecord.SetSeverityNumber(5)
+					case "DEBUG2":
+						logRecord.SetSeverityNumber(6)
+					case "DEBUG3":
+						logRecord.SetSeverityNumber(7)
+					case "DEBUG4":
+						logRecord.SetSeverityNumber(8)
+					case "INFO":
+						logRecord.SetSeverityNumber(9)
+					case "INFO2":
+						logRecord.SetSeverityNumber(10)
+					case "INFO3":
+						logRecord.SetSeverityNumber(11)
+					case "INFO4":
+						logRecord.SetSeverityNumber(12)
+					case "WARN":
+						logRecord.SetSeverityNumber(13)
+					case "WARN2":
+						logRecord.SetSeverityNumber(14)
+					case "WARN3":
+						logRecord.SetSeverityNumber(15)
+					case "WARN4":
+						logRecord.SetSeverityNumber(16)
+					case "ERROR":
+						logRecord.SetSeverityNumber(17)
+					case "ERROR2":
+						logRecord.SetSeverityNumber(18)
+					case "ERROR3":
+						logRecord.SetSeverityNumber(19)
+					case "ERROR4":
+						logRecord.SetSeverityNumber(20)
+					case "FATAL":
+						logRecord.SetSeverityNumber(21)
+					case "FATAL2":
+						logRecord.SetSeverityNumber(22)
+					case "FATAL3":
+						logRecord.SetSeverityNumber(23)
+					case "FATAL4":
+						logRecord.SetSeverityNumber(24)
+					default:
+					}
+					logRecord.SetSeverityText(logRecord.SeverityNumber().String())
+				}
+				if requestId, ok := record["requestId"].(string); ok {
+					logRecord.Attributes().PutStr(semconv.AttributeFaaSInvocationID, requestId)
+				}
+				if line, ok := record["message"].(string); ok {
+					logRecord.Body().SetStr(line)
+				}
+			} else {
+				// in plain text https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#telemetry-api-function
+				if line, ok := el.Record.(string); ok {
+					logRecord.Body().SetStr(line)
+				}
+			}
+		}
+	}
+	return log, nil
+}
+
+func (r *telemetryAPIReceiver) registerTracesConsumer(next consumer.Traces) {
+	r.nextTraces = next
```

Comment:
since we are trying to reuse the http server and it seems that we want to have a single instance of traces consume and logs consumer, would it make sense for `nextTraces` to be an array?

This would allow to have multiple receivers in different pipelines.

Reply:
My thinking is 1 telemetry API receiver can have 1 traces consumer and 1 logs consumer.
I made some changes so as to support multiple instances of telemetry API receiver.
If we want to have multiple receivers in different pipelines, I would suggest supporting it using multiple instances of telemetry API receivers.